sympy/solvers/solveset.py

```diff
@@ -121,6 +121,31 @@ def invert_real(f_x, y, x, domain=S.Reals):
     """
     return _invert(f_x, y, x, domain)
 
+
+def power(x, y):
+    """
+    Returns an integer if `y` can be expressed in terms of power of `x`,
+    otherwise `False`.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import power
+    >>> power(2, 8)
+    3
```

Comment:
Is this any better in general? (Caution: lightly tested.)

See also `integer_nthroot` and `perfect_power`
```
>>> def power(x, y):
...     bey = perfect_power(y)
...     if bey:
...         if x == bey[0]: return bey[1]
...         bex = perfect_power(x)
...         if bex[0] == bey[0]:
...             ans, rem = divmod(bey[1], bex[1])
...             if not rem:
...                 return ans
...     return False
... 
>>> power(9, 81)
2
>>> power(9, 81*3)
False
```

wait...this is `multiplicity`

```
def power(x, y):
    i = multiplicity(x, y)
    if not i: return False
    return i
```